server/handlers.go

```diff
@@ -749,6 +759,10 @@ func (s *Server) handleToken(w http.ResponseWriter, r *http.Request) {
 		}
 		return
 	}
+	if clientSecret == "" && client.Secret != "" && r.PostFormValue("code_verifier") != "" {
```

Comment:
Seems inconsistent that we want to specifically call out "missing credentials" in PKCE case and not in general case. 
Also I'm wondering whether not specifying client authentication error details but just general "Invalid client credentials." is for security reasons. E.g. the error you introduced would allow to determine whether the client having a specific client_id is registered on the server.

Reply:
I understand. Should we instead just log it?

Reply:
I would say it's a  good idea, many such entries could indicate that some attack is in place. We could also log when client_id is invalid. Not sure if we would like to specifically call-out "missing credentials" case, but I'm not against it. And let's do it regardless of PKCE.

Reply:
Ok, Now, there is a log output instead.

Reply:
I was thinking more in lines of:

```
           if client.Secret != clientSecret {
	        if clientSecret != "" {
	            s.logger.Info("missing client secret")
	        } else {
	            s.logger.Info("invalid client secret")
	        }
	        
		s.tokenErrHelper(w, errInvalidClient, "Invalid client credentials.", http.StatusUnauthorized)					 
                return
```

Reply:
Written text has so much potential for misunderstanding ;o)